gyp/pylib/gyp/generator/compile_commands_json.py

```diff
@@ -81,7 +81,7 @@ def resolve(filename):
         commands = per_config_commands.setdefault(configuration_name, [])
         for source in sources:
             file = resolve(source)
-            isc = source.endswith(".c")
+            isc = source.endswith(".c") or source.endswith(".m")
```

Comment:
```suggestion
            isc = source.endswith((".c", ".m"))
```
https://docs.astral.sh/ruff/rules/multiple-starts-ends-with